### ISSUE TITLE
[V3] Implement new Concept Exercise: advanced-errors

### ISSUE THREAD

**Issue by tehsphinx** [exercism/go#1375]
This issue describes how to implement the `advanced-errors` concept exercise for the Go track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

The goal of this exercise is to teach the student about `custom error types` and `error wrapping` in Go.

## Learning objectives

- Implement custom error types (any type with an `Error() string` function can be used as an error)
- Wrapping/Unwrapping of errors
- Check for specific errors (`errors.Is`, `errors.As`)

## Out of scope

- error handling
- panic/recover

## Concepts

- `error-wrapping`
- `error-unwrapping`
- `custom-errors`
- `error-type-check`

## Prerequisites

- `errors`
- `interfaces`

## Resources to refer to

- https://blog.golang.org/go1.13-errors

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.
